eclipse-scout-core/test/menu/menusSpec.js

```diff
@@ -105,6 +107,27 @@ describe('menus', () => {
       expect(items).toEqual([]);
     });
 
+    it('returns a menu with no menuTypes if the at least one defaultMenuType matches the given types', () => {
+      let menu1, menu2, menu3;
+      menu1 = helper.createMenu(helper.createModel(1));
+      menu1.menuTypes = ['test.MenuType1', 'test.MenuType2'];
+      menu1.visible = false;
+      menu2 = helper.createMenu(helper.createModel(2));
+      menu2.menuTypes = ['test.MenuType1'];
+      menu3 = helper.createMenu(helper.createModel(3));
+      menu3.menuTypes = ['test.MenuType2'];
```

Comment:
Maybe we should add a setMenuTypes function on Menu. Even though it probably won't have an effect if the menu is already rendered in the menubar, it will at least trigger a property change event which could be used to handle that case (no need to do that now)